### PR TITLE
Support for --force for firewall rules.

### PR DESCRIPTION
**Pull request by G-Harmon**
cc @csbell @nikhiljindal @madhusudancs

---
This change is [Reviewable](https://reviewable.io/reviews/googlecloudplatform/k8s-multicluster-ingress/77)

### CODE REVIEW

File: app/kubemci/pkg/gcp/firewallrule/firewallrulesyncer.go [hunk 1]
```diff
@@ -128,11 +138,33 @@ func (s *FirewallRuleSyncer) createFirewallRule(desiredFR *compute.Firewall) err
...
+	existingFR.CreationTimestamp = ""
+	existingFR.Id = 0
+	existingFR.Kind = ""
+	existingFR.SelfLink = ""
+	// Check for and clear default Direction.
+	if existingFR.Direction == "INGRESS" {
+		existingFR.Direction = ""
+	}
+	// Check for and clear the default Priority.
+	if existingFR.Priority == 1000 {
```
Comment (nikhiljindal) [inline app/kubemci/pkg/gcp/firewallrule/firewallrulesyncer.go]: hmm why is this required?
This will break if the default is changed. Should we care?

Comment (G-Harmon) [reply app/kubemci/pkg/gcp/firewallrule/firewallrulesyncer.go]: The Firewall Rules that come back from gcp include the priority. We have 2 choices for dealing with this fact: 1) add P=1000 to our "desired" firewall rule; or 2) ignore P==1000 in our comparison. It seems pretty similar either way.
When the default changes:
with method 1), we'll be using the old default.
with method 2), we'll consider the rules not to match. And then until we fix the code, we'll always have to use --force, or a future "update" command.
So method (2) seems slightly better in that we won't use "old" defaults for anything. (Note that previous commits have been using method (1).)

Comment (nikhiljindal) [reply app/kubemci/pkg/gcp/firewallrule/firewallrulesyncer.go]: Another option is to ignore priority (dont compare it)?

Comment (G-Harmon) [reply app/kubemci/pkg/gcp/firewallrule/firewallrulesyncer.go]: We discussed this in-person. I wrote up our thoughts in Issue#86. Basically, we'll move important attributes to the 'desiredBlah' function, and we'll decide what attributes can be ignored on a case-by-case basis, and put those into the BlahMatches function.
Pushing a new commit shortly.

**Comment by G-Harmon** [comment]
PTAL- I added in a bug fix in targetproxysyncer. Without that fix, creating the MCI fails.

**Comment by nikhiljindal** [comment]
Assuming you have verified that "make vet fmt" still passes.

**Comment by G-Harmon** [comment]
thanks. yes, vet and fmt are still clean.
